### PR TITLE
feat: reduce code (& stack) size

### PR DESCRIPTION
**Pull request by giuseppe998e**
The execution time of tests done with the previous version and with these modifications are about `1.800` (`s.ms`) (on my i5) in both cases.
In contrast, the space occupied by the code and the memory allocated on the stack are smaller (_in my opinion_, it could be useful if the library is used in webapps written, for example, with frameworks like React).

Signed-off-by: Giuseppe Eletto <peppe.eletto@gmail.com>

### CODE REVIEW

**Comment by sindresorhus** [comment]
While I appreciate the pull request, I don't really see the point in code golfing the code. Readability and maintainability is an important factor too.

**Comment by sindresorhus** [comment]
> The execution time of tests done with the previous version and with these modifications are about 1.800 (s.ms) (on my i5) in both cases.

Performance could vary with different real-world usages, machines, Node.js versions, etc. Your changes could just as easily worsen performance for the majority of use-cases instead of increasing it.

**Comment by sindresorhus** [comment]
> In contrast, the space occupied by the code and the memory allocated on the stack are smaller

The savings in minified & gzipped code is minuscule. 4 tiny static arrays are not gonna have any impact.

File: index.js [hunk 1]
```diff
@@ -1,68 +1,19 @@
...
-	'b',
-	'kibit',
-	'Mibit',
-	'Gibit',
-	'Tibit',
-	'Pibit',
-	'Eibit',
-	'Zibit',
-	'Yibit'
-];
```
Comment (sindresorhus) [inline index.js:49]: We could potentially turn these into `Set` which does have a tiny bit faster look-up time in theory.

**Comment by giuseppe998e** [comment]
**From my point of view**, it is useless to allocate 4 different arrays on the stack (permanently as they are outside the function) and then use only one at a time, especially when the difference between them is so little.

Actually my PR had to stop at the 2nd commit (`fix: follow repo coding style`), the following ones I made for myself, I didn't know they were added automatically in this PR.

I'm glad to have received a reply and I absolutely don't want to insist on the result of my work on the code.

**Comment by sindresorhus** [comment]
> From my point of view, it is useless to allocate 4 different arrays on the stack and then use only one at a time, especially when the difference between them is so little.

Good code is a balance of readability, performance, memory usage, and code size. In this case, I would argue readability is more important than saving the memory usage of 4 tiny arrays.

File: index.js [hunk 2]
```diff
@@ -72,47 +23,43 @@ module.exports = (number, options) => {
...
-	const exponent = Math.min(Math.floor(options.binary ? Math.log(number) / Math.log(1024) : Math.log10(number) / 3), UNITS.length - 1);
+	const exponent = Math.min(
+		Math.floor(options.binary ? Math.log(number) / Math.log(1024) : Math.log10(number) / 3),
+		UNITS_FIRSTLETTER.length - 1
+	);
 	// eslint-disable-next-line unicorn/prefer-exponentiation-operator
 	number /= Math.pow(options.binary ? 1024 : 1000, exponent);
 
-	if (!localeOptions) {
+	if (!Object.keys(localeOptions).length) {
```
Comment (sindresorhus) [inline index.js:52]: I'm happy to take this and the other object checks if you wanna do a separate pull request though.

Comment (giuseppe998e) [reply index.js:52]: @sindresorhus, these object controls are tied to lines 35-38 (of my code) which replaced 90-98 (of the project).
Because the project variable `localOptions`, if empty, will be `undefined` on "your" code, but returns an empty object on "mine".
So to change these checks I should also include lines 35-38.

Comment (sindresorhus) [reply index.js:52]: Ah, missed that. Never mind then.
